app-modules/student-data-model/src/Filament/Resources/StudentResource/Pages/ViewStudent.php

```diff
@@ -165,9 +171,45 @@ public function infolist(Infolist $infolist): Infolist
 
     public function sisRefresh()
     {
+        [$olympusUrl, $olympusKey] = Landlord::execute(function (): array {
+            $settings = app(OlympusSettings::class);
+
+            return [
+                rtrim($settings->url, '/'),
+                $settings->key,
+            ];
+        });
+
+        $tenantId = Tenant::current()->getKey();
+
+        /** @var Student $student */
+        $student = $this->getRecord();
+
+        try {
+            $response = Http::withToken($olympusKey)
+                ->asJson()
+                ->post("{$olympusUrl}/integrations/{$tenantId}/student-on-demand-sync", [
+                    'sisid' => $student->getKey(),
+                    'otherid' => $student->otherid,
+                ])
+                ->throw();
```

Comment:
The code looks good, but I would suggest putting the actual request in an action class so it can be swapped out for a mock when testing (without needing to fake the entire Http facade). The Filament notification bit is UI-related so should stay in this Livewire component

Reply:
Like I said above, I think I want to consider in the near future actually creating a SisIntegration concept for Advising App that would be called here instead. But I would like to start at a small increment like this before going much further to get the Feature out.

Knowing that, given how small the portion here is for the HTTP request, and how easy it is to mock fake the response I would propose leaving this as it is for the time being.

What do you think?